source/crc32.c

```diff
@@ -0,0 +1,72 @@
+#include <aws/cal/crc32.h>
+#include <aws/cal/hash.h>
+#include <aws/checksums/crc.h>
+
+/* hash is object, vtable is methods, hash properties (not vtable) are data?*/
+
+static void s_destroy(struct aws_hash *hash);
+static int s_update(struct aws_hash *hash, const struct aws_byte_cursor *to_hash);
+static int s_finalize(struct aws_hash *hash, struct aws_byte_buf *output);
+/* why do we need to define and implement the functions separatly? */
+/* can we reuse some of these as defaults defined in hash.c?*/
+
+/* does this have to be defined globaly? I'd like to use a different update function depending on a parameter */
+static struct aws_hash_vtable s_vtable = {
+    .destroy = s_destroy,
+    .update = s_update,
+    .finalize = s_finalize,
+    .alg_name = "CRC32",
+    .provider = "AWS", /* is this right? */
+};
+
+struct aws_hash *aws_crc32_default_new(struct aws_allocator *allocator) {
+    struct aws_hash *hash = aws_mem_acquire(allocator, sizeof(struct aws_hash));
+
+    if (!hash) {
+        return NULL;
+    }
+
+    hash->allocator = allocator;
+    hash->vtable = &s_vtable; /*could make to vtables, and pass an argument to default new*/
+    hash->impl = 0;
+    hash->digest_size = AWS_CRC32C_LEN;
+    hash->good = true;
+
+    return hash;
+}
+
+/* reordered from destroy update finalize to update finalize destroy, this order makes more sense to me */
```

Comment:
This comment doesn't match the code, can probably go?

Reply:
This is in relation to the order the methods show up in Sha256CC. It doesn't really matter, but I think update finalize destroy makes more sense since I imagine that's the order the functions would be invoked during the lifetime of the object.

Reply:
Someone else coming in to read this code later will not have just read sha256.c most likely. You want to make it so that someone coming in fresh reads your comment and understands the following code. This would just confuse them :)